File: Alura/while.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

console.log("*********************************");
console.log("Bem vindo no jogo de adivinhação!");
console.log("*********************************");

const numeroSecreto = 42;
const totalDeTentativa = 3;
let rodada = 1;

function realizarRodada() {
  console.log(`Tentativa ${rodada} de ${totalDeTentativa}`);
  rl.question("Digite o seu número: ", (chuteStr) => {
    console.log(`Você digitou ${chuteStr}`);
    const chute = parseInt(chuteStr);
  
    const acertou = chute === numeroSecreto;
    const maior = chute > numeroSecreto;
    const menor = chute < numeroSecreto;
  
    if (acertou) {
      console.log("Você acertou");
      rl.close();
    } else {
      if (maior) {
        console.log("Você errou! O seu chute foi maior que o número secreto.");
      } else if (menor) {
        console.log("Você errou! O seu chute foi menor do que o número secreto.");
      }
      rodada++;
      if (rodada <= totalDeTentativa) {
        realizarRodada();
      } else {
        console.log("Fim do jogo");
        rl.close();
      }
    }
  });
}

realizarRodada();

File: Alura/while.py
print("*********************************")
print("Bem vindo no jogo de adivinhação!")
print("*********************************")

numero_secreto = 42
total_de_tentativa = 3
rodada = 1

while(rodada <= total_de_tentativa):
  #print("Tentativa", rodada, "de", total_de_tentativa)
  print("Tentativa {} de {}". format(rodada, total_de_tentativa))
  chute_str = input("Digite o seu número: ")
  print("Você digitou ", chute_str)
  chute = int(chute_str)
  
  acertou = chute == numero_secreto
  maior = chute > numero_secreto
  menor = chute < numero_secreto

  if (acertou):
    print("Você acertou")
  else:
    if(maior):
      print("Você errou! O seu chute foi maior que o número secreto.")
    elif(menor):
      print("Você errou! O seu chute foi menor do que o número secreto.")
    
  rodada = rodada + 1
  
print("Fim do jogo")